plugins/typescript/src/templates/fetcher.ts

```diff
@@ -77,11 +77,12 @@ export async function ${camel(prefix)}Fetch<
       {
         signal,
         method: method.toUpperCase(),
-        body: body ? JSON.stringify(body) : undefined,
-        headers: {
-          "Content-Type": "application/json",
-          ...headers,
-        },
+        body: body ? (body instanceof FormData ? body : JSON.stringify(body)) : undefined,
+        headers: headers
+          ? headers
+          : {
+              "Content-Type": "application/json"
```

Comment:
looks like we are losing the default `"Content-Type": "application/json"` if we don't pass headers.

Reply:
If headers isn't passed then the default will still apply. 

This was to work around the boundary information missing when we do a multipart form data post. Apparently if you leave the content-type empty the browser will supply the correct content-type header with the correct boundary.

Unsure of a better solution here - if the boundary isn't supplied, the recieving backend can fail to understand the request properly.

https://stackoverflow.com/questions/39280438/fetch-missing-boundary-in-multipart-form-data-post

Reply:
On second thoughts it may be possible to supply a custom boundary - so we can keep the default content type and override if needed for multi parts.

https://stackoverflow.com/questions/3508338/what-is-the-boundary-in-multipart-form-data 

I will investigate tomorrow! :)

Reply:
cool 👍

Reply:
@needim - I've made an update that hopefully keeps the best of both worlds.

Essentially when using the fetch api. if multipart/form-data is specified it breaks when the browser sends the request. This is because the browser automatically adds its own boundary (needed for this type of request)

See the warning specified here:
https://developer.mozilla.org/en-US/docs/Web/API/FormData/Using_FormData_Objects#sending_files_using_a_formdata_object

> Warning: When using FormData to submit POST requests using [XMLHttpRequest](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest) or the [Fetch_API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) with the multipart/form-data Content-Type (e.g. when uploading Files and Blobs to the server), do not explicitly set the [Content-Type](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type) header on the request. Doing so will prevent the browser from being able to set the Content-Type header with the boundary expression it will use to delimit form fields in the request body.